dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
 
-FROM docker:${DOCKER_VERSION}-dind-rootless
+FROM golang:alpine AS build
+WORKDIR /build
+COPY docker-iptables-glue/go.* ./
+RUN go mod download
+COPY docker-iptables-glue/ ./
+RUN CGO_ENABLED=0 GOOS=linux go build
```

Comment:
It is not a good idea to hardcode `GOOS`, as it would break compatibility with other OSes in the future. It's recommended that you trim the build path from the binary and statically compile it. You might also be interested in enabling or striping debug symbols to enable or prohibit remote debugging.

Take a look at this Dockerfile for Go: https://github.com/inloco/artifactcache/blob/master/Dockerfile, it is a good example of the things I've pointed above.

Reply:
a572e3a